### PR TITLE
Do not download build or flash if not actually needed

### PR DESCRIPTION
**Pull request by FranciscoCano**

### CODE REVIEW

File: scripts/get_latest_build.py [hunk 1]
```diff
@@ -80,6 +74,13 @@ def main():
     f = detect_build_file(options.source, options.username, options.passwd, options.device, options.build_type,
                           options.branch, last_date)
 
+    # If there is no build file for the given date, it can mean it has not been generated yet,
```
Comment (villenavg) [inline scripts/get_latest_build.py]: Why the change?

Comment (FranciscoCano) [inline scripts/get_latest_build.py]: Because if there was no new build available, the current one in the device was being downloaded and flashed.

Comment (villenavg) [inline scripts/get_latest_build.py]: all right

**Comment by villenavg** [comment]
LGTM
